android/src/com/mapswithme/maps/widget/placepage/SponsoredHotel.java

```diff
@@ -73,6 +173,15 @@ static void requestPrice(String id, String currencyCode)
     nativeRequestPrice(id, currencyCode);
   }
 
+  static void requestInfo(String id, String locale)
+  {
+    HotelInfo info = sInfoCache.get(id);
+    if (info != null)
+      onInfoReceived(id, info);
+
+    nativeRequestInfo(id, locale);
```

Comment:
> onInfoReceived is called from the C++ code
As far as I understand, `onInfoReceived` method is called from the managed code (Java code) from the `SponsoredHotel.requestInfo` method. Why did you say that method is called from the C++? Could you tell with more details what is call chain in this case and what is the root method which called from the C++ ?